codice/ddf
# DDF-3719 Updates UI build to ease extensibility

#### What does this PR do?

- Consolidates all webpack.config.js into single `ace bundle` command
- Bumps webpack to latest version `4.5.0`
- Updates webpack plugins/loaders to work with latest webpack version
- Updates UI modules to always clean target directories before building
- Refactors imports to reduce the need for special configuration
- Refactors `spec-loader` and `concat-less` loader to be more generic

#### Who is reviewing it? 

@mackncheesiest 
@adimka 

#### Select relevant component teams: 

@codice/ui 
@codice/build 

#### Choose 2 committers to review/merge the PR. 

@andrewkfiedler
@pklinef

#### How should this be tested?

- Full build of `ui` directory on osx, windows, linux
- Ensure `yarn start` continues to work on osx, windows, linux
- Ensure the `catalog-ui-search` ui has no regressions

#### What are the relevant tickets?

[DDF-3719](https://codice.atlassian.net/browse/DDF-3719)

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
